### PR TITLE
Implement web based tests

### PR DESCRIPTION
**Pull request by mikehaertl**
An alternative test implementation.

 * Tests via HTTP and FastCGI
 * New tests only require update of `env` in `.travis.yml`
 * Requirements for each flavour can be specified in `tests/requirements-<flavour>.php`

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -2,25 +2,32 @@ language: bash
...
+  - VERSION=7.1 VARIANT=fpm TYPE=fcgi
+  - VERSION=7.0 VARIANT=apache TYPE=http
+  - VERSION=7.0 VARIANT=fpm TYPE=fcgi
+  - VERSION=5.6 VARIANT=apache TYPE=http
+  - VERSION=5.6 VARIANT=fpm TYPE=fcgi
 
-install: true
+install:
+  - git clone https://github.com/yiisoft/yii2.git ~/build/yiisoft/yii2-docker/tests/yii2
+  - sudo apt-get install libfcgi0ldbl
```
Comment (schmunk42) [inline .travis.yml]: What's the purpose of installing this on the Travis machine?

File: tests/run.sh [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+elif [ $3 = "common" ]; then
+    QUERY='flavour=common'
+else
+    echo >&2 "Invalid flavour $3"
+    exit 1
+fi
+
+
+if [ $2 = "http" ]; then
+    ID=$(docker run -d -p 80:80 -v $PWD:/var/www/html --name test $1)
```
Comment (schmunk42) [inline tests/run.sh]: How do you get a webserver on port 80 running?

Comment (mikehaertl) [inline tests/run.sh]: Not sure what you mean. The default port for Apache images is port 80.

File: .travis.yml (hunk 1 above)
Comment (mikehaertl) [inline .travis.yml]: It provides the `fcgi-bin` that we need to test FastCGI based images (PHP-FPM)

File: tests/run.sh (hunk 2 above)
Comment (schmunk42) [inline tests/run.sh]: Yeah, was in the fpm-only world since a while, sorry :)

**Pull request merged**